### PR TITLE
Adding EXPERIMENTAL native support for InfluxDB

### PR DESCRIPTION
**Pull request by TFyre**
Adding EXPERIMENTAL native support for InfluxDB

Also fix native image build

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -54,4 +54,4 @@ jobs:
       run: mvn -B formatter:validate install --file pom.xml
 
     - name: Build in native
-      run: mvn -B -Pnative verify --file integration-tests/pom.xml
+      run: mvn -B -Pnative-image verify --file integration-tests/pom.xml
```
Comment (ebullient) [inline .github/workflows/build.yml:57]: You're right that this is wrong, but this is the wrong fix. Please use the `native` property to enable the profile: `-Dnative`

**Review by ebullient — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (TFyre) [reply .github/workflows/build.yml]: 👍

File: micrometer-registry-influx/deployment/src/main/java/io/quarkiverse/micrometer/registry/influx/deployment/InfluxRegistryProcessor.java [hunk 2]
```diff
@@ -36,9 +38,20 @@ public boolean getAsBoolean() {
     }
 
     @BuildStep(onlyIf = { NativeBuild.class, InfluxRegistryEnabled.class })
-    public MicrometerRegistryProviderBuildItem nativeModeNotSupported() {
-        log.info("The InfluxDB meter registry does not support running in native mode.");
-        return null;
+    public MicrometerRegistryProviderBuildItem nativeModeNotSupported(
```
Comment (ebullient) [inline micrometer-registry-influx/deployment/src/main/java/io/quarkiverse/micrometer/registry/influx/deployment/InfluxRegistryProcessor.java:41]: maybe a method name change. ;)

File: micrometer-registry-influx/deployment/src/main/java/io/quarkiverse/micrometer/registry/influx/deployment/InfluxRegistryProcessor.java [hunk 3]
```diff
@@ -36,9 +38,20 @@ public boolean getAsBoolean() {
...
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans,
+            BuildProducer<ReflectiveClassBuildItem> reflectionClasses) {
+        log.warn("native support for InfluxDB meter registery is EXPERIMENTAL!!! check reflectionClasses");
+
+        reflectionClasses.produce(new ReflectiveClassBuildItem(true, true, InfluxApiVersion.class));
+
+        // Add the InfluxDB Registry Producer
```
Comment (ebullient) [inline micrometer-registry-influx/deployment/src/main/java/io/quarkiverse/micrometer/registry/influx/deployment/InfluxRegistryProcessor.java:48]: You should be able to remove the "onlyIfNot" clause off of the next method.

Constrain this method adding the reflection classes required for native..

File: micrometer-registry-influx/deployment/src/main/java/io/quarkiverse/micrometer/registry/influx/deployment/InfluxRegistryProcessor.java (hunk 2 above)
Comment (TFyre) [reply micrometer-registry-influx/deployment/src/main/java/io/quarkiverse/micrometer/registry/influx/deployment/InfluxRegistryProcessor.java]: whooops!

**Review by ebullient — APPROVED** [review]
🎉

**Pull request merged**